### PR TITLE
[#5965] subtask(web): ML model support for web UI

### PR DESCRIPTION
**Pull request by LauraXia123**
### What changes were proposed in this pull request?
create catalog
image

create schema
image

register/view/drop/list model
image
image

link/view/drop/list versions
image
image
image
image

### Why are the changes needed?

(Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.)

Fix: #5817

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
manually

### CODE REVIEW

**Comment by jerryshao** [comment]
Can you please also add the web UI doc in this PR.

**Comment by LauraXia123** [comment]
Sure, I updated the commit.

**Comment by jerryshao** [comment]
The CI is failed, can you please fix the CI, thanks.

File: docs/assets/webui/link-version.png
Comment (jerryshao) [inline docs/assets/webui/link-version.png:1]: Why this image cannot be displayed?

File: docs/assets/webui/list-columns.png
Comment (jerryshao) [inline docs/assets/webui/list-columns.png:1]: Is this image correct?

File: docs/webui.md [hunk 1]
```diff
@@ -160,11 +160,12 @@ Click on the `CREATE CATALOG` button displays the dialog to create a catalog.
 Creating a catalog requires these fields:
 
 1. **Catalog name**(**_required_**): the name of the catalog
-2. **Type**(**_required_**): `relational`/`fileset`/`messaging`, the default value is `relational`
+2. **Type**(**_required_**): `relational`/`fileset`/`messaging`/`model`, the default value is `relational`
 3. **Provider**(**_required_**):
     1. Type `relational` - `hive`/`iceberg`/`mysql`/`postgresql`/`doris`/`paimon`/`hudi`/`oceanbase`
     2. Type `fileset` - `hadoop`
     3. Type `messaging` - `kafka`
+    4. Type `model` have no provider
```
Comment (jerryshao) [inline docs/webui.md:168]: "...has no provider"

File: docs/webui.md [hunk 2]
```diff
@@ -468,15 +469,62 @@ Displays a confirmation dialog, clicking on the `DROP` button drops this schema.
...
+
+![create-table](./assets/webui/create-table.png)
+
+Creating a table needs these fields:
+
+1. **Name**(**_required_**): the name of the table.
+2. **columns**(**_required_**): 
+    1. The name and type of each column are required.
+    2. Only suppport simple types, cannot support complex types by ui, you can create complex types by api.
+3. **Comment**(_optional_): the comment of the fileset.
```
Comment (jerryshao) [inline docs/webui.md:490]: "...the comment of the table."?

File: docs/webui.md [hunk 3]
```diff
@@ -527,19 +575,137 @@ Displays a confirmation dialog, clicking on the `DROP` button drops this fileset
...
+#### Drop model
+
+Click on the action icon <Icon icon='mdi:delete-outline' fontSize='24' color='red' /> in the table cell.
+
+Displays a confirmation dialog, clicking on the `DROP` button drops this model.
+
+![delete-model](./assets/webui/delete-model.png)
+
+### Version
+Click the model tree node on the left sidebar or the model name link in the table cell.
```
Comment (jerryshao) [inline docs/webui.md:662]: Can you add a blank line above?

File: docs/webui.md [hunk 4]
```diff
@@ -527,19 +575,137 @@ Displays a confirmation dialog, clicking on the `DROP` button drops this fileset
...
+
+Displays the dialog for modifying fields of the selected topic.
+
+![update-topic-dialog](./assets/webui/update-topic-dialog.png)
+
+#### Drop topic
+
+Click on the action icon <Icon icon='mdi:delete-outline' fontSize='24' color='red' /> in the table cell.
+
+Displays a confirmation dialog, clicking on the `DROP` button drops this fileset.
```
Comment (orenccl) [inline docs/webui.md:622]: ... drops this topic.

File: docs/webui.md (hunk 1 above)
Comment (justinmclean) [reply docs/webui.md:168]: If it has no provider, then provider is also not required?

File: web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+    setInnerProps(data)
+    setValue('propItems', data)
+  }
+
+  const addFields = () => {
+    const duplicateKeys = innerProps
+      .filter(item => item.key.trim() !== '')
+      .some(
+        (item, index, filteredItems) =>
+          filteredItems.findIndex(otherItem => otherItem !== item && otherItem.key.trim() === item.key.trim()) !== -1
```
[the remark this answers is not in the document]
Comment (LauraXia123) [reply web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js]: Sure, thanks for the offer.

File: web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+            linkVersion({ data: schemaData, metalake, catalog, schema: schemaName, type: catalogType, model })
+          ).then(res => {
+            if (!res.payload?.err) {
+              handleClose()
+            }
+          })
+        }
+      })
+      .catch(err => {
+        console.error('valid error', err)
```
[the remark this answers is not in the document]
Comment (LauraXia123) [reply web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js:233]: This is just for debugging purposes.

File: web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+                            control={control}
+                            render={({ field }) => (
+                              <TextField
+                                {...field}
+                                onChange={event => {
+                                  field.onChange(event)
+                                  trigger('aliases')
+                                }}
+                                label={`Aliase ${index + 1}`}
+                                error={!!errors.aliases?.[index]?.name || !!errors.aliases?.message}
```
[the remark this answers is not in the document]
Comment (LauraXia123) [reply web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js:330]: The error field accepts a boolean value

File: docs/webui.md (hunk 1 above)
Comment (LauraXia123) [reply docs/webui.md:168]: Required for type `relational /fileset/messaging`.

File: web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (orenccl) [reply web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js:233]: Got it.

File: web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (orenccl) [reply web/web/src/app/metalakes/metalake/rightContent/LinkVersionDialog.js:330]: Got it. Thanks.

**Review by orenccl — APPROVED** [review]
~~deleted~~

**Review by orenccl — COMMENTED** [review]
docs/assets/webui/list-columns.png is actually correct. Need to click view file to see full image.

image

**Review by orenccl — APPROVED** [review]

**Comment by orenccl** [comment]
Hi @LauraXia123,
Did you forget to upload the updated docs/webui.md file in the "Improve Docs and Code" commit?

File: docs/assets/webui/list-columns.png
Comment (orenccl) [reply docs/assets/webui/list-columns.png:1]: docs/assets/webui/list-columns.png is actually correct. Need to click view file to see full image.

**Comment by LauraXia123** [comment]
> Hi @LauraXia123, Did you forget to upload the updated docs/webui.md file in the "Improve Docs and Code" commit?

sorry, I missed it.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**